New examples

jesrui opened the pull request:
These are ports of some of the lessons of  [Will Usher's SDL tutorials](http://www.willusher.io/pages/sdl2/) to rust.

jdeseno commented:
Hey thanks, I appreciate taking the effort to improve the examples.

Licenses/copyrights: please convert `res/ice-troll.bmp` to what you need or get sign off on the original images. Ditto for droidsans.ttf but, SDL_ttf isn't part of this project so I think that example and dependency can be removed.

`README.md` needs a description of what each example does and how they're different from the other image examples, currently they seem a bit redundant.

There's an endless number of things to do with images in SDL2, these examples should be extremely simple and not overly dependent on other knowledge e.g. "put an image on the screen", "handle a mouse click".

Please rebase as well.

jdeseno commented:
Thinking about it for a second more, I think 2 examples "How to load images using SDL_image", "How to display text using SDL_ttf" would be 2 good examples you could extract here.

jesrui commented:
Ok, I have updated the PR:

- License: both this and Will Usher's SDL tutorials have a MIT license.
- Images: I have converted a copy of `res/ice-troll.bmp` to PNG  format with a transparent background.
- Font: SansDroid replaced with a public domain TTF
- I have taken out two of the examples as they were a bit repetitive.
- README updated
- A bit of code & comment clean up
